plugins/pulp_rpm/plugins/migrations/0020_nested_drpm_directory.py

```diff
@@ -0,0 +1,119 @@
+import glob
+import logging
+import os
+import shutil
+import tempfile
+
+import lxml.etree as le
+
+from pulp.server.config import config
+from pulp.server.db.connection import get_collection
+
+
+_logger = logging.getLogger('pulp_rpm.plugins.migrations.0020')
+
+
+def _move_nested_drpm_dir(repo_dir):
+    """
+    If a repository has a nested <repo_dir>/drpm/drpm/ directory,
+    remove one of the nestings as it should not have been created
+    :param repo_dir: The root directory of the repository
+    :type repo_dir: str
+    """
+    drpm_dir = os.path.join(repo_dir, 'drpms', 'drpms')
+    target_drpm_dir = os.path.join(repo_dir, 'drpms')
+    if os.path.exists(drpm_dir):
+        _logger.debug('Cleaning double drpm dir: %s' % drpm_dir)
+        tempdir = tempfile.mkdtemp(dir=repo_dir)
+        shutil.move(drpm_dir, tempdir)
+        shutil.rmtree(target_drpm_dir)
+        shutil.move(os.path.join(tempdir, 'drpms'), repo_dir)
+        shutil.rmtree(tempdir)
+
+
+def _remove_prestodelta_repo_units():
+    """
+    Remove all prestodelta repo_content_units since they should nto have been created
+    to begin with
+    """
+    metadata_collection = get_collection('units_yum_repo_metadata_file')
+    repo_units_collection = get_collection('repo_content_units')
+    for presto_unit in metadata_collection.find({'data_type': 'prestodelta'}):
+        # remove any repo repo units that reference it, the unit itself will
+        # be removed by the orphan cleanup at some point in the future
```

Comment:
Because if we clean right away then the migration needs to modify the content directory when it can do nothing and the content will be removed normally